pkg/auth/login_credentials_manager.go

```diff
@@ -323,6 +371,20 @@ func (h *LoginCredentialsManagerImpl) isSSOUser(email, organizationId string) bo
 	return err == nil && res.GetIsSso()
 }
 
+func (h *LoginCredentialsManagerImpl) isOnPremSSOUser(url, caCertPath string, unsafeTrace bool) bool {
+	clientManager := &MDSClientManagerImpl{}
+	client, err := clientManager.GetMDSClient(url, caCertPath, unsafeTrace)
+	if err != nil {
+		return false
+	}
+
+	featuresInfo, _, err := client.MetadataServiceOperationsApi.Features(context.Background())
+	if err != nil {
+		return false
+	}
+	return featuresInfo.Features["oidc.login.device.1.enabled"]
```

Comment:
Let's check if this exists so we don't panic if it doesn't?
```suggestion
	enabled, ok := featuresInfo.Features["oidc.login.device.1.enabled"]
```

Reply:
It doesn't panic if the key doesn't exist, it just returns the zero value (so `false` in this case since `Features` is `map[string]bool`). Tested in go playground.

Reply:
TIL!